vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java

```diff
@@ -58,10 +64,13 @@ public String space(@PathVariable("id") String id, Model model) {
             model.addAttribute("space", space);
             model.addAttribute("moduleList", moduleLinkManager.getLinkDisplays(id));
             if(space.isShowUnpublishedLinks()) {
-                model.addAttribute("spaceLinks",spaceLinkManager.getLinkDisplays(id));
+                spaceLinks = spaceLinkManager.getLinkDisplays(id);
+                
             }else {
-                model.addAttribute("spaceLinks", spaceLinkManager.getSpaceLinkForGivenOrNullSpaceStatus(id, SpaceStatus.PUBLISHED));
+            	spaceLinks = spaceLinkManager.getSpaceLinkForGivenOrNullSpaceStatus(id, SpaceStatus.PUBLISHED);                
             }
+            List<ISpaceLinkDisplay> filteredSpaceLinks = spaceLinks.stream().filter(spaceLinkDisplayObj -> !spaceLinkDisplayObj.getLink().getTargetSpace().getHideIncomingLinks() || spaceLinkDisplayObj.getLink().getSourceSpace().getSpaceStatus().equals(SpaceStatus.UNPUBLISHED)).collect(Collectors.toList());
```

Comment:
what is this for:
` spaceLinkDisplayObj.getLink().getSourceSpace().getSpaceStatus().equals(SpaceStatus.UNPUBLISHED)`
if `space.isShowUnpublishedLinks()` is true, the one you're filtering out should be shown, otherwise they would already be filtered out.

Reply:
If there are 2 spaces defined say Space 1, with Space ID: ```1001``` and Space 2 with Space ID: ```1002```, and I make Space 2 as ```Unpublished``` and Space 1 is ```Published```. 
When I go to the public view page of Space 1, I am checking the incoming space links space status, i.e. I am checking whether there is an incoming link from Space 2 on Space 1 and if the status of Space 2 is ```Unpublished```, I filter those links out, and it's not shown Space 1 public view page.

Reply:
but what is this for 
`spaceLinkDisplayObj.getLink().getSourceSpace().getSpaceStatus().equals(SpaceStatus.UNPUBLISHED)`
source space is the current space, isn't it? so if that wasn't published, then it wouldn't even get to this line.

Reply:
Yes, my bad, I should change it to ```spaceLinkDisplayObj.getLink().getSourceSpace().getTargetSpace().equals(SpaceStatus.UNPUBLISHED)```

Reply:
that is taken care of in line 66/67, isn't it?

Reply:
But that line, I suppose you mean the below line:
```spaceLinks = spaceLinkManager.getSpaceLinkForGivenOrNullSpaceStatus(id, SpaceStatus.PUBLISHED);```
Here, ```PUBLISHED``` status is being passed, but I am checking if the target space is ```UNPUBLISHED``` by default hide all the links.

Reply:
No, I mean:
```
if(space.isShowUnpublishedLinks()) {
               spaceLinks = spaceLinkManager.getLinkDisplays(id);
```
Only if a space has `showUnpublishedLinks` set to false, it should hide all links to unpublished spaces if I'm not mistaken. By adding another check for the status of the linked space you're overriding what this code is doing.

Reply:
No it's should not override the code, because in the stream logic, I check if my flag variable ```hideIncomingLinks``` is set to ```True``` or ```False```. 
The existing code:
```if(space.isShowUnpublishedLinks()) {```
               ```spaceLinks = spaceLinkManager.getLinkDisplays(id);```
checks if other space linked to the given space is unpublished, if yes it shows all the links, and for that I do another check in the stream logic to check if ```hideIncomingLinks``` is set to ```True``` or ```False```.

Reply:
Shall I re-open PR?

Reply:
your stream logic is to collect all links where both of these equal false, right?
`!spaceLinkDisplayObj.getLink().getTargetSpace().getHideIncomingLinks() || spaceLinkDisplayObj.getLink().getTargetSpace().getSpaceStatus().equals(SpaceStatus.UNPUBLISHED)`

so, if `spaceLinks` has 1 link (L1) to an unpublished space (S1) and 1 link (L2) to a published space (S2),
if S1.hideIncomingLink = False it would be shown
if S2.hideIncomingLink = False it would not be shown because that would be equivalent to 
`false || false` which is `false`, so it wouldn't be shown
right? Or am I missing something? why do you need the second condition (`spaceLinkDisplayObj.getLink().getTargetSpace().getSpaceStatus().equals(SpaceStatus.UNPUBLISHED)`)?

Reply:
I am using the second condition because, by default -> if space is set to ```UNPUBLISHED``` I need to hide all the links right, irrespective of the option for hide/unhide links provided that sets ```hideIncomingLinks``` to ```TRUE/FALSE```. Hence, the second condition is to check whether the space is ```UNPUBLISHED/PUBLISHED```.

Reply:
no, this is what lines 66-71 should take care of

Reply:
So, should I remove the second condition then?

Reply:
yes, unless there is a valid reason why it should be there.